### PR TITLE
SELC-1197 new api retrieve billing

### PR DESCRIPTION
**Pull request by antonioT90**
added new API to retrieve Billing data even if the current user has not relation to the current party/product

### CODE REVIEW

**Review by TonioGela — APPROVED** [review]

File: src/test/scala/it/pagopa/interop/partyprocess/ExternalApiSpec.scala [hunk 1]
```diff
@@ -254,4 +256,131 @@ trait ExternalApiSpec
...
+                      billing = PartyManagementDependency.Billing(
+                        vatNumber = "VATNUMBER",
+                        recipientCode = "RECIPIENTCODE",
+                        publicServices = Option(true)
+                      )
+                    )
+                  )
+                )
+            )
+          else Future.failed(ResourceNotFoundError(externalId))
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/partyprocess/ExternalApiSpec.scala]: personal taste, but I'd suggest to keep the mocks simple to improve readability.
I would favour having multiple mock functions without parameters, suck as `mockRetrieveInstitutionNotFound`, `mockRetrieveInstitutionWithoutBilling` and `mockRetrieveInstitutionWithBilling`.

But it's really just a mild suggestion 😄

**Review by galales — APPROVED** [review]

File: src/test/scala/it/pagopa/interop/partyprocess/ExternalApiSpec.scala (hunk 1 above)
Comment (antonioT90) [reply src/test/scala/it/pagopa/interop/partyprocess/ExternalApiSpec.scala]: this is a java vs scala developer question I think 🤣
I'l change it 😉

File: src/main/scala/it/pagopa/interop/partyprocess/api/impl/ExternalApiServiceImpl.scala [hunk 2]
```diff
@@ -179,4 +179,43 @@ class ExternalApiServiceImpl(
...
+   */
+  override def getBillingInstitutionByExternalId(externalId: String, productId: String)(implicit
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerBillingData: ToEntityMarshaller[BillingData],
+    contexts: Seq[(String, String)]
+  ): Route = {
+    logger.info(s"Retrieving billing data for institution having externalId $externalId and productId $productId")
+    val result = for {
+      bearer      <- getFutureBearer(contexts)
+      _           <- getUidFuture(contexts)
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/partyprocess/api/impl/ExternalApiServiceImpl.scala]: Which is the purpose of this `getUidFuture`?

Comment (antonioT90) [reply src/main/scala/it/pagopa/interop/partyprocess/api/impl/ExternalApiServiceImpl.scala]: @beetlecrunch I started copying from another API:
https://github.com/pagopa/interop-be-party-process/blob/SELC-1197_new_api_retrieve_billing/src/main/scala/it/pagopa/interop/partyprocess/api/impl/ExternalApiServiceImpl.scala#L45

I was thinking that it was checking something from the current context.
Maybe I should remove it also from the API linked above?

Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/partyprocess/api/impl/ExternalApiServiceImpl.scala]: Ok, I suppose that it is a typo.
Let's remove it.

**Review by galales — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/partyprocess/api/impl/ExternalApiServiceImpl.scala (hunk 2 above)
Comment (antonioT90) [reply src/main/scala/it/pagopa/interop/partyprocess/api/impl/ExternalApiServiceImpl.scala]: removed from each point in which it was used as showed

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**